uwrt_mars_rover_utils/uwrt_mars_rover_estop/src/estop.cpp

```diff
@@ -0,0 +1,73 @@
+#include <uwrt_mars_rover_estop/estop.hpp>
+
+namespace uwrt_mars_rover_estop{
+    Estop::Estop(const rclcpp::NodeOptions &options) : Node("estop_button", options){
+        RCLCPP_INFO(this->get_logger(),"ESTOP NODE HAS STARTED");
+
+        //create publisher for middle man topic
+        cmd_vel_publisher_ = this->create_publisher<geometry_msgs::msg::Twist>("estop_vel_interceptor",10);
+        
+        //callback for checking if the enter key has been pressed
+        auto estop_callback = [this](const std_msgs::msg::Bool::SharedPtr msg) -> void {
+            
+            //if enter is presed then the estop toggle will be enabled
+            if(msg->data){
+                isEstop = true;
+            }
+
+        
+                                
+        };
+
+
+        //callback for the subscriber will read the values for the drivetrain topic
+        //we check the state of the estop boolean here
+        auto cmd_subscriber_callback = [this](const geometry_msgs::msg::Twist::SharedPtr msg) -> void {
```

Comment:
I think we can simplify this to writing to a /robot_state topic that will either be "running" or "estop" right now. If the enter key is pressed, then it will publish "estop". If not, it will publish running. The drivetrain node will check the state and will read /cmd_vel if "running" but will set velocity to 0 if "estop" and stop listening to cmd_vel. What do you think @keyonjerome ? Also if we do follow this route, we should probably define a new ros message called robot_state where we define a bunch of constants like ROBOT_STATE_RUNNING = 0, ROBOT_STATE_ESTOP = 1 so that the message is actually an int message that just uses those constants we define.